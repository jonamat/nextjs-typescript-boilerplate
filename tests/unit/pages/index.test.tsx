/* ----------------------------- Unit test suite ---------------------------- *
 *
 * Target: Entry page
 * Tests:
 * - Behavior
 * - Consistency
 *
 * -------------------------------------------------------------------------- */

import React from 'react';
import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';

import Home from '@/pages/index';

/* -------------------------------------------------------------------------- */
/*                                    Mocks                                   */
/* -------------------------------------------------------------------------- */

/* -------------------------------------------------------------------------- */
/*                              Bridge operations                             */
/* -------------------------------------------------------------------------- */

beforeEach(() => {
  jest.clearAllMocks();
});

/* -------------------------------------------------------------------------- */
/*                              Wrapper function                              */
/* -------------------------------------------------------------------------- */

const run = (props?) => {
  return render(<Home {...props} />);
};

/* -------------------------------------------------------------------------- */
/*                                    Tests                                   */
/* -------------------------------------------------------------------------- */

describe('Behavior', () => {
  it('renders without crash', () => {
    expect(() => {
      run();
    }).not.toThrowError();
  });
});

describe('Consistency', () => {
  it('matches the snapshot', () => {
    const tree = create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
