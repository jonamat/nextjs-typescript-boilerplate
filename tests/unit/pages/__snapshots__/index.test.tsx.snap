// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page matches the snapshot 1`] = `
<div
  className="css-h74a88"
>
  <div
    className="css-1878w0a"
  >
    <h2
      className="chakra-heading css-1r2f04i"
    >
      NextJS boilerplate
    </h2>
    <button
      className="chakra-button css-1xx8uq7"
      onClick={[Function]}
      onMouseEnter={[Function]}
      type="button"
    >
      Go to countries page
    </button>
  </div>
</div>
`;
